app/controllers/departments_controller.rb

```diff
@@ -1,8 +1,8 @@
 class DepartmentsController < ApplicationController
+  before_action :authenticate_user!
```

Comment:
hàm này làm gì e

Reply:
authenticate_user!: là method đảm bảo người dùng đã đăng nhập. Hàm này được gọi thông qua before_filter. 

- Nếu như user chưa đăng nhập, như mặc dịnh nó sẽ bị redirect về trang login page. Nếu như user đăng nhập thành công, nó sẽ về trang root mà bạn đã cài đặt trong routes.
- Sử dụng authenticate_user! đảm bảo việc user đăng nhập nên ta hoàn toàn có thể sử dụng method current_user để truy cập database hay thao tác trên nó:
- Before_filter :authenticate_user! đảm bảo current_user sẽ không bao giờ bị nil